Source/SwiftyDropbox/Platform/SwiftyDropbox_iOS/OAuthMobile.swift

```diff
@@ -78,21 +114,35 @@ open class DropboxMobileOAuthManager: DropboxOAuthManager {
     }
 
     fileprivate func dAuthURL(_ scheme: String, nonce: String?) -> URL {
-        var components = URLComponents()
-        components.scheme =  scheme
-        components.host = "1"
-        components.path = "/connect"
-        
+        var components = dauthUrlCommonComponents(with: scheme)
         if let n = nonce {
             let state = "oauth2:\(n)"
-            components.queryItems = [
-                URLQueryItem(name: "k", value: self.appKey),
-                URLQueryItem(name: "s", value: ""),
-                URLQueryItem(name: "state", value: state),
-            ]
+            components.queryItems?.append(URLQueryItem(name: OAuthConstants.stateKey, value: state))
         }
         return components.url!
     }
+
+    private func dAuthURL(_ scheme: String, authSession: AuthSession) -> URL {
+        var components = dauthUrlCommonComponents(with: scheme)
+        let extraQueryParams = Self.createExtraQueryParamsString(for: authSession)
+        components.queryItems?.append(contentsOf: [
+            URLQueryItem(name: OAuthConstants.stateKey, value: authSession.state),
+            URLQueryItem(name: OAuthConstants.extraQueryParamsKey, value: extraQueryParams),
+        ])
+        return components.url!
```

Comment:
We shouldnt force unwrap. I'd prefer us to either:

A: Allow the caller to handle the nullable URL
B: guard against null url and fatalError here.